// @coreui/angular/esm2020/lib/footer/app-footer.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { AppFooterComponent } from './app-footer.component';
import * as i0 from "@angular/core";
export class AppFooterModule {
}
AppFooterModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.0.5", ngImport: i0, type: AppFooterModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AppFooterModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.0.5", ngImport: i0, type: AppFooterModule, declarations: [AppFooterComponent], imports: [CommonModule], exports: [AppFooterComponent] });
AppFooterModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.0.5", ngImport: i0, type: AppFooterModule, imports: [CommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.0.5", ngImport: i0, type: AppFooterModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    exports: [AppFooterComponent],
                    declarations: [AppFooterComponent]
                }]
        }] });
