// @coreui/angular/esm2020/lib/sidebar/app-sidebar-header.component.mjs
import { Component, HostBinding } from '@angular/core';
import * as i0 from "@angular/core";
export class AppSidebarHeaderComponent {
    constructor() {
        this.sidebarHeaderClass = true;
    }
}
AppSidebarHeaderComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.0.5", ngImport: i0, type: AppSidebarHeaderComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
AppSidebarHeaderComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "14.0.5", type: AppSidebarHeaderComponent, selector: "app-sidebar-header, cui-sidebar-header", host: { properties: { "class.sidebar-header": "this.sidebarHeaderClass" } }, ngImport: i0, template: `<ng-content></ng-content>`, isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.0.5", ngImport: i0, type: AppSidebarHeaderComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'app-sidebar-header, cui-sidebar-header',
                    template: `<ng-content></ng-content>`
                }]
        }], ctorParameters: function () { return []; }, propDecorators: { sidebarHeaderClass: [{
                type: HostBinding,
                args: ['class.sidebar-header']
            }] } });
