// @coreui/angular/esm2020/lib/sidebar/app-sidebar-nav.directive.mjs
import { Directive, HostListener } from '@angular/core';
import * as i0 from "@angular/core";
export class NavDropdownDirective {
    constructor(el) {
        this.el = el;
    }
    toggle() {
        this.el.nativeElement.classList.toggle('open');
    }
}
NavDropdownDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.0.5", ngImport: i0, type: NavDropdownDirective, deps: [{ token: i0.ElementRef }], target: i0.ɵɵFactoryTarget.Directive });
NavDropdownDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "14.0.5", type: NavDropdownDirective, selector: "[appNavDropdown]", ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.0.5", ngImport: i0, type: NavDropdownDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[appNavDropdown]'
                }]
        }], ctorParameters: function () { return [{ type: i0.ElementRef }]; } });
/**
 * Allows the dropdown to be toggled via click.
 */
export class NavDropdownToggleDirective {
    constructor(dropdown) {
        this.dropdown = dropdown;
    }
    toggleOpen($event) {
        $event.preventDefault();
        this.dropdown.toggle();
    }
}
NavDropdownToggleDirective.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.0.5", ngImport: i0, type: NavDropdownToggleDirective, deps: [{ token: NavDropdownDirective }], target: i0.ɵɵFactoryTarget.Directive });
NavDropdownToggleDirective.ɵdir = i0.ɵɵngDeclareDirective({ minVersion: "14.0.0", version: "14.0.5", type: NavDropdownToggleDirective, selector: "[appNavDropdownToggle]", host: { listeners: { "click": "toggleOpen($event)" } }, ngImport: i0 });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.0.5", ngImport: i0, type: NavDropdownToggleDirective, decorators: [{
            type: Directive,
            args: [{
                    selector: '[appNavDropdownToggle]'
                }]
        }], ctorParameters: function () { return [{ type: NavDropdownDirective }]; }, propDecorators: { toggleOpen: [{
                type: HostListener,
                args: ['click', ['$event']]
            }] } });
