// @coreui/angular/esm2020/lib/sidebar/app-sidebar-footer.component.mjs
import { Component, HostBinding } from '@angular/core';
import * as i0 from "@angular/core";
export class AppSidebarFooterComponent {
    constructor() {
        this.sidebarFooterClass = true;
    }
}
AppSidebarFooterComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.0.5", ngImport: i0, type: AppSidebarFooterComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
AppSidebarFooterComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "14.0.5", type: AppSidebarFooterComponent, selector: "app-sidebar-footer, cui-sidebar-footer", host: { properties: { "class.sidebar-footer": "this.sidebarFooterClass" } }, ngImport: i0, template: `<ng-content></ng-content>`, isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.0.5", ngImport: i0, type: AppSidebarFooterComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'app-sidebar-footer, cui-sidebar-footer',
                    template: `<ng-content></ng-content>`
                }]
        }], ctorParameters: function () { return []; }, propDecorators: { sidebarFooterClass: [{
                type: HostBinding,
                args: ['class.sidebar-footer']
            }] } });
