// @coreui/angular/esm2020/lib/sidebar/app-sidebar-form.component.mjs
import { Component, HostBinding } from '@angular/core';
import * as i0 from "@angular/core";
export class AppSidebarFormComponent {
    constructor() {
        this.sidebarFormClass = true;
    }
}
AppSidebarFormComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.0.5", ngImport: i0, type: AppSidebarFormComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
AppSidebarFormComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "14.0.5", type: AppSidebarFormComponent, selector: "app-sidebar-form, cui-sidebar-form", host: { properties: { "class.sidebar-form": "this.sidebarFormClass" } }, ngImport: i0, template: `<ng-content></ng-content>`, isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.0.5", ngImport: i0, type: AppSidebarFormComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'app-sidebar-form, cui-sidebar-form',
                    template: `<ng-content></ng-content>`
                }]
        }], ctorParameters: function () { return []; }, propDecorators: { sidebarFormClass: [{
                type: HostBinding,
                args: ['class.sidebar-form']
            }] } });
