// @coreui/angular/esm2020/lib/sidebar/app-sidebar-minimizer.component.mjs
import { Component, HostBinding, HostListener, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "./app-sidebar.service";
export class AppSidebarMinimizerComponent {
    constructor(sidebarService) {
        this.sidebarService = sidebarService;
        this.role = 'button';
        this.sidebarMinimizerClass = true;
    }
    toggleOpen($event) {
        $event.preventDefault();
        this.sidebarService.toggle({ minimize: 'toggle' });
    }
}
AppSidebarMinimizerComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.0.5", ngImport: i0, type: AppSidebarMinimizerComponent, deps: [{ token: i1.AppSidebarService }], target: i0.ɵɵFactoryTarget.Component });
AppSidebarMinimizerComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "14.0.5", type: AppSidebarMinimizerComponent, selector: "app-sidebar-minimizer, cui-sidebar-minimizer", inputs: { role: "role" }, host: { listeners: { "click": "toggleOpen($event)" }, properties: { "attr.role": "this.role", "class.sidebar-minimizer": "this.sidebarMinimizerClass" } }, ngImport: i0, template: ``, isInline: true });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.0.5", ngImport: i0, type: AppSidebarMinimizerComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'app-sidebar-minimizer, cui-sidebar-minimizer',
                    template: ``
                }]
        }], ctorParameters: function () { return [{ type: i1.AppSidebarService }]; }, propDecorators: { role: [{
                type: HostBinding,
                args: ['attr.role']
            }, {
                type: Input
            }], sidebarMinimizerClass: [{
                type: HostBinding,
                args: ['class.sidebar-minimizer']
            }], toggleOpen: [{
                type: HostListener,
                args: ['click', ['$event']]
            }] } });
