// @coreui/angular/esm2020/lib/sidebar/app-sidebar-nav.component.mjs
import { Component, HostBinding, Input } from '@angular/core';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
import * as i2 from "./app-sidebar-nav/app-sidebar-nav-items.component";
export class AppSidebarNavComponent {
    constructor(router) {
        this.router = router;
        this.navItems = [];
        this.sidebarNavClass = true;
        this.role = 'nav';
        this.navItemsArray = [];
    }
    ngOnChanges(changes) {
        this.navItemsArray = Array.isArray(this.navItems) ? this.navItems.slice() : [];
    }
}
AppSidebarNavComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.0.5", ngImport: i0, type: AppSidebarNavComponent, deps: [{ token: i1.Router }], target: i0.ɵɵFactoryTarget.Component });
AppSidebarNavComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "14.0.5", type: AppSidebarNavComponent, selector: "app-sidebar-nav, cui-sidebar-nav", inputs: { navItems: "navItems", role: "role" }, host: { properties: { "class.sidebar-nav": "this.sidebarNavClass", "attr.role": "this.role" } }, usesOnChanges: true, ngImport: i0, template: "<app-sidebar-nav-items\n  class=\"nav\"\n  [items]=\"navItemsArray\">\n</app-sidebar-nav-items>\n", dependencies: [{ kind: "component", type: i2.AppSidebarNavItemsComponent, selector: "app-sidebar-nav-items, cui-sidebar-nav-items", inputs: ["items"] }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.0.5", ngImport: i0, type: AppSidebarNavComponent, decorators: [{
            type: Component,
            args: [{ selector: 'app-sidebar-nav, cui-sidebar-nav', template: "<app-sidebar-nav-items\n  class=\"nav\"\n  [items]=\"navItemsArray\">\n</app-sidebar-nav-items>\n" }]
        }], ctorParameters: function () { return [{ type: i1.Router }]; }, propDecorators: { navItems: [{
                type: Input
            }], sidebarNavClass: [{
                type: HostBinding,
                args: ['class.sidebar-nav']
            }], role: [{
                type: HostBinding,
                args: ['attr.role']
            }, {
                type: Input
            }] } });
