// @coreui/angular/esm2020/lib/aside/app-aside.module.mjs
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { LayoutModule } from '../shared/layout/layout.module';
import { AppAsideComponent } from './app-aside.component';
import * as i0 from "@angular/core";
export class AppAsideModule {
}
AppAsideModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.0.5", ngImport: i0, type: AppAsideModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
AppAsideModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "14.0.5", ngImport: i0, type: AppAsideModule, declarations: [AppAsideComponent], imports: [CommonModule,
        LayoutModule], exports: [AppAsideComponent] });
AppAsideModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "14.0.5", ngImport: i0, type: AppAsideModule, imports: [CommonModule,
        LayoutModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.0.5", ngImport: i0, type: AppAsideModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule,
                        LayoutModule
                    ],
                    exports: [
                        AppAsideComponent
                    ],
                    declarations: [
                        AppAsideComponent
                    ]
                }]
        }] });
