// @coreui/angular/esm2020/lib/sidebar/app-sidebar-nav/app-sidebar-nav-badge.pipe.mjs
import { Pipe } from '@angular/core';
import * as i0 from "@angular/core";
export class AppSidebarNavBadgePipe {
    transform(item, args) {
        const classes = {
            badge: true
        };
        const variant = `badge-${item.badge.variant}`;
        classes[variant] = !!item.badge.variant;
        classes[item.badge.class] = !!item.badge.class;
        return classes;
    }
}
AppSidebarNavBadgePipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.0.5", ngImport: i0, type: AppSidebarNavBadgePipe, deps: [], target: i0.ɵɵFactoryTarget.Pipe });
AppSidebarNavBadgePipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "14.0.5", ngImport: i0, type: AppSidebarNavBadgePipe, name: "appSidebarNavBadge" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.0.5", ngImport: i0, type: AppSidebarNavBadgePipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'appSidebarNavBadge'
                }]
        }] });
