// @coreui/angular/esm2020/lib/sidebar/app-sidebar-nav/app-sidebar-nav-link.pipe.mjs
import { Pipe } from '@angular/core';
import * as i0 from "@angular/core";
export class AppSidebarNavLinkPipe {
    transform(item) {
        const classes = { 'nav-link': true };
        const disabled = item.attributes && item.attributes.disabled;
        classes['disabled'] = disabled;
        classes['btn-link'] = disabled;
        classes[`nav-link-${item.variant}`] = !!item.variant;
        return classes;
    }
}
AppSidebarNavLinkPipe.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.0.5", ngImport: i0, type: AppSidebarNavLinkPipe, deps: [], target: i0.ɵɵFactoryTarget.Pipe });
AppSidebarNavLinkPipe.ɵpipe = i0.ɵɵngDeclarePipe({ minVersion: "14.0.0", version: "14.0.5", ngImport: i0, type: AppSidebarNavLinkPipe, name: "appSidebarNavLink" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.0.5", ngImport: i0, type: AppSidebarNavLinkPipe, decorators: [{
            type: Pipe,
            args: [{
                    name: 'appSidebarNavLink'
                }]
        }] });
