// @coreui/angular/esm2020/lib/sidebar/app-sidebar-nav/app-sidebar-nav-items.component.mjs
import { Component, Inject, Input } from '@angular/core';
import { DOCUMENT } from '@angular/common';
import { SidebarNavHelper } from '../app-sidebar-nav.service';
import * as i0 from "@angular/core";
import * as i1 from "@angular/router";
import * as i2 from "../app-sidebar-nav.service";
import * as i3 from "@angular/common";
import * as i4 from "../../shared/layout/layout.directive";
import * as i5 from "./app-sidebar-nav-divider.component";
import * as i6 from "./app-sidebar-nav-link.component";
import * as i7 from "./app-sidebar-nav-title.component";
import * as i8 from "../app-sidebar-nav.directive";
import * as i9 from "./app-sidebar-nav-label.component";
import * as i10 from "./app-sidebar-nav-item-class.pipe";
import * as i11 from "./app-sidebar-nav-icon.pipe";
import * as i12 from "./app-sidebar-nav-badge.pipe";
export class AppSidebarNavItemsComponent {
    constructor(document, renderer, router, helper) {
        this.document = document;
        this.renderer = renderer;
        this.router = router;
        this.helper = helper;
        this._items = [];
    }
    set items(items) {
        this._items = [...items];
    }
    get items() {
        return this._items;
    }
    hideMobile() {
        if (this.document.body.classList.contains('sidebar-show')) {
            this.renderer.removeClass(this.document.body, 'sidebar-show');
        }
    }
}
AppSidebarNavItemsComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.0.5", ngImport: i0, type: AppSidebarNavItemsComponent, deps: [{ token: DOCUMENT }, { token: i0.Renderer2 }, { token: i1.Router }, { token: i2.SidebarNavHelper }], target: i0.ɵɵFactoryTarget.Component });
AppSidebarNavItemsComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "14.0.5", type: AppSidebarNavItemsComponent, selector: "app-sidebar-nav-items, cui-sidebar-nav-items", inputs: { items: "items" }, ngImport: i0, template: `
    <ng-container *ngFor="let item of items">
      <ng-container [ngSwitch]="helper.itemType(item)">
        <app-sidebar-nav-dropdown
          *ngSwitchCase="'dropdown'"
          [item]="item"
          [class.open]="helper.isActive(router, item)"
          [ngClass]="item | appSidebarNavItemClass"
          appNavDropdown
          routerLinkActive="open">
        </app-sidebar-nav-dropdown>
        <app-sidebar-nav-divider
          *ngSwitchCase="'divider'"
          [item]="item"
          [ngClass]="item | appSidebarNavItemClass"
          [appHtmlAttr]="item.attributes">
        </app-sidebar-nav-divider>
        <app-sidebar-nav-title
          *ngSwitchCase="'title'"
          [item]="item"
          [ngClass]="item | appSidebarNavItemClass"
          [appHtmlAttr]="item.attributes">
        </app-sidebar-nav-title>
        <app-sidebar-nav-label
          *ngSwitchCase="'label'"
          [item]="item"
          class="nav-item"
          [ngClass]="item | appSidebarNavItemClass">
        </app-sidebar-nav-label>
        <ng-container
          *ngSwitchCase="'empty'">
        </ng-container>
        <app-sidebar-nav-link
          *ngSwitchDefault
          [item]="item"
          class="nav-item"
          [ngClass]="item | appSidebarNavItemClass"
          (linkClick)="hideMobile()"
        >
        </app-sidebar-nav-link>
      </ng-container>
    </ng-container>
  `, isInline: true, dependencies: [{ kind: "directive", type: i0.forwardRef(function () { return i3.NgClass; }), selector: "[ngClass]", inputs: ["class", "ngClass"] }, { kind: "directive", type: i0.forwardRef(function () { return i3.NgForOf; }), selector: "[ngFor][ngForOf]", inputs: ["ngForOf", "ngForTrackBy", "ngForTemplate"] }, { kind: "directive", type: i0.forwardRef(function () { return i3.NgSwitch; }), selector: "[ngSwitch]", inputs: ["ngSwitch"] }, { kind: "directive", type: i0.forwardRef(function () { return i3.NgSwitchCase; }), selector: "[ngSwitchCase]", inputs: ["ngSwitchCase"] }, { kind: "directive", type: i0.forwardRef(function () { return i3.NgSwitchDefault; }), selector: "[ngSwitchDefault]" }, { kind: "directive", type: i0.forwardRef(function () { return i1.RouterLinkActive; }), selector: "[routerLinkActive]", inputs: ["routerLinkActiveOptions", "ariaCurrentWhenActive", "routerLinkActive"], outputs: ["isActiveChange"], exportAs: ["routerLinkActive"] }, { kind: "directive", type: i0.forwardRef(function () { return i4.HtmlAttributesDirective; }), selector: "[appHtmlAttr]", inputs: ["appHtmlAttr"] }, { kind: "component", type: i0.forwardRef(function () { return i5.AppSidebarNavDividerComponent; }), selector: "app-sidebar-nav-divider, cui-sidebar-nav-divider", inputs: ["item"] }, { kind: "component", type: i0.forwardRef(function () { return AppSidebarNavDropdownComponent; }), selector: "app-sidebar-nav-dropdown, cui-sidebar-nav-dropdown", inputs: ["item"] }, { kind: "component", type: i0.forwardRef(function () { return i6.AppSidebarNavLinkComponent; }), selector: "app-sidebar-nav-link, cui-sidebar-nav-link", inputs: ["item"], outputs: ["linkClick"] }, { kind: "component", type: i0.forwardRef(function () { return i7.AppSidebarNavTitleComponent; }), selector: "app-sidebar-nav-title, cui-sidebar-nav-title", inputs: ["item"] }, { kind: "directive", type: i0.forwardRef(function () { return i8.NavDropdownDirective; }), selector: "[appNavDropdown]" }, { kind: "component", type: i0.forwardRef(function () { return i9.AppSidebarNavLabelComponent; }), selector: "app-sidebar-nav-label, cui-sidebar-nav-label", inputs: ["item"] }, { kind: "pipe", type: i0.forwardRef(function () { return i10.AppSidebarNavItemClassPipe; }), name: "appSidebarNavItemClass" }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.0.5", ngImport: i0, type: AppSidebarNavItemsComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'app-sidebar-nav-items, cui-sidebar-nav-items',
                    template: `
    <ng-container *ngFor="let item of items">
      <ng-container [ngSwitch]="helper.itemType(item)">
        <app-sidebar-nav-dropdown
          *ngSwitchCase="'dropdown'"
          [item]="item"
          [class.open]="helper.isActive(router, item)"
          [ngClass]="item | appSidebarNavItemClass"
          appNavDropdown
          routerLinkActive="open">
        </app-sidebar-nav-dropdown>
        <app-sidebar-nav-divider
          *ngSwitchCase="'divider'"
          [item]="item"
          [ngClass]="item | appSidebarNavItemClass"
          [appHtmlAttr]="item.attributes">
        </app-sidebar-nav-divider>
        <app-sidebar-nav-title
          *ngSwitchCase="'title'"
          [item]="item"
          [ngClass]="item | appSidebarNavItemClass"
          [appHtmlAttr]="item.attributes">
        </app-sidebar-nav-title>
        <app-sidebar-nav-label
          *ngSwitchCase="'label'"
          [item]="item"
          class="nav-item"
          [ngClass]="item | appSidebarNavItemClass">
        </app-sidebar-nav-label>
        <ng-container
          *ngSwitchCase="'empty'">
        </ng-container>
        <app-sidebar-nav-link
          *ngSwitchDefault
          [item]="item"
          class="nav-item"
          [ngClass]="item | appSidebarNavItemClass"
          (linkClick)="hideMobile()"
        >
        </app-sidebar-nav-link>
      </ng-container>
    </ng-container>
  `
                }]
        }], ctorParameters: function () { return [{ type: undefined, decorators: [{
                    type: Inject,
                    args: [DOCUMENT]
                }] }, { type: i0.Renderer2 }, { type: i1.Router }, { type: i2.SidebarNavHelper }]; }, propDecorators: { items: [{
                type: Input
            }] } });
export class AppSidebarNavDropdownComponent {
    constructor(helper) {
        this.helper = helper;
        this.item = {};
    }
}
AppSidebarNavDropdownComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "14.0.5", ngImport: i0, type: AppSidebarNavDropdownComponent, deps: [{ token: i2.SidebarNavHelper }], target: i0.ɵɵFactoryTarget.Component });
AppSidebarNavDropdownComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "14.0.0", version: "14.0.5", type: AppSidebarNavDropdownComponent, selector: "app-sidebar-nav-dropdown, cui-sidebar-nav-dropdown", inputs: { item: "item" }, providers: [SidebarNavHelper], ngImport: i0, template: `
    <a class="nav-link nav-dropdown-toggle"
       appNavDropdownToggle
       [appHtmlAttr]="item.attributes">
      <i *ngIf="helper.hasIcon(item)" [ngClass]="item | appSidebarNavIcon"></i>
      <ng-container>{{item.name}}</ng-container>
      <span *ngIf="helper.hasBadge(item)" [ngClass]="item | appSidebarNavBadge">{{ item.badge?.text }}</span>
    </a>
    <app-sidebar-nav-items
      class="nav-dropdown-items"
      [items]="item.children ?? []">
    </app-sidebar-nav-items>
  `, isInline: true, styles: [".nav-dropdown-toggle{cursor:pointer}\n", ".nav-dropdown-items{display:block}\n"], dependencies: [{ kind: "directive", type: i3.NgClass, selector: "[ngClass]", inputs: ["class", "ngClass"] }, { kind: "directive", type: i3.NgIf, selector: "[ngIf]", inputs: ["ngIf", "ngIfThen", "ngIfElse"] }, { kind: "directive", type: i4.HtmlAttributesDirective, selector: "[appHtmlAttr]", inputs: ["appHtmlAttr"] }, { kind: "component", type: AppSidebarNavItemsComponent, selector: "app-sidebar-nav-items, cui-sidebar-nav-items", inputs: ["items"] }, { kind: "directive", type: i8.NavDropdownToggleDirective, selector: "[appNavDropdownToggle]" }, { kind: "pipe", type: i11.AppSidebarNavIconPipe, name: "appSidebarNavIcon" }, { kind: "pipe", type: i12.AppSidebarNavBadgePipe, name: "appSidebarNavBadge" }] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "14.0.5", ngImport: i0, type: AppSidebarNavDropdownComponent, decorators: [{
            type: Component,
            args: [{ selector: 'app-sidebar-nav-dropdown, cui-sidebar-nav-dropdown', template: `
    <a class="nav-link nav-dropdown-toggle"
       appNavDropdownToggle
       [appHtmlAttr]="item.attributes">
      <i *ngIf="helper.hasIcon(item)" [ngClass]="item | appSidebarNavIcon"></i>
      <ng-container>{{item.name}}</ng-container>
      <span *ngIf="helper.hasBadge(item)" [ngClass]="item | appSidebarNavBadge">{{ item.badge?.text }}</span>
    </a>
    <app-sidebar-nav-items
      class="nav-dropdown-items"
      [items]="item.children ?? []">
    </app-sidebar-nav-items>
  `, providers: [SidebarNavHelper], styles: [".nav-dropdown-toggle{cursor:pointer}\n", ".nav-dropdown-items{display:block}\n"] }]
        }], ctorParameters: function () { return [{ type: i2.SidebarNavHelper }]; }, propDecorators: { item: [{
                type: Input
            }] } });
